cura/Settings/MachineManager.py

```diff
@@ -306,12 +349,25 @@ def setActiveMachine(self, stack_id: str) -> None:
 
         self.__emitChangedSignals()
 
+    ##  Given a definition id, return the machine with this id.
+    #   Optional: add a list of keys and values to filter the list of machines with the given definition id
+    #   \param definition_id \type{str} definition id that needs to look for
+    #   \param metadata_filter \type{dict} list of metadata keys and values used for filtering
     @staticmethod
-    def getMachine(definition_id: str) -> Optional["GlobalStack"]:
+    def getMachine(definition_id: str, metadata_filter: Dict[str, str] = None) -> Optional["GlobalStack"]:
         machines = ContainerRegistry.getInstance().findContainerStacks(type = "machine")
         for machine in machines:
             if machine.definition.getId() == definition_id:
-                return machine
+                if metadata_filter:
+                    pass_all_filters = True
+                    for key in metadata_filter:
```

Comment:
actually I think you can let findContainerStacks do this extra filtering as well. In that case you just pass the metadata_filter along

Reply:
Yes, I was thinking about it and at the end I did it differently. Changed and done in 180139090a085be88ce4aa2e8b5ff2712589917f